testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/ejb/DefaultElytronEjbSecurityDomainTestCase.java

```diff
@@ -0,0 +1,149 @@
+ * Test case on a deployment containing a secured EJB with non-default security domain and an unsecured one.
+ * Test passes if deployment is successful and functional.
+ */
+@RunWith(Arquillian.class)
+@ServerSetup(DefaultElytronEjbSecurityDomainTestCase.DefaultElytronEjbSecurityDomainTestCaseSecuritySetup.class)
+public class DefaultElytronEjbSecurityDomainTestCase {
+
+    @ArquillianResource
+    private Context ctx;
+
+    @EJB(mappedName = "java:global/ejb3security/WhoAmIBean!org.wildfly.test.integration.elytron.ejb.WhoAmI")
+    private WhoAmI securedBean;
+
+    @Deployment
+    public static JavaArchive createDeployment() {
+        final Package currentPackage = DefaultElytronEjbSecurityDomainTestCase.class.getPackage();
+        return ShrinkWrap.create(JavaArchive.class, "ejb-security-domain-test.jar")
+                .addPackage(DefaultElytronEjbSecurityDomainTestCase.class.getPackage())
+                .addPackage(EjbUnsecuredBean.class.getPackage())
+                .addPackage(org.wildfly.test.integration.elytron.ejb.base.WhoAmIBean.class.getPackage())
+                .addPackage(EntryBean.class.getPackage())
+                .addClass(Util.class)
+                .addClass(WhoAmI.class)
+                .addClass(org.jboss.as.controller.operations.common.Util.class)
+                .addPackage(AbstractSecurityDomainSetup.class.getPackage())
+                .addAsResource(currentPackage, "users.properties", "users.properties")
+                .addAsResource(currentPackage, "roles.properties", "roles.properties")
+                .addAsManifestResource(currentPackage, "jboss-ejb3.xml", "jboss-ejb3.xml")
+                .addAsManifestResource(createPermissionsXmlAsset(new ElytronPermission("getSecurityDomain")), "permissions.xml");
+    }
+
+    @Test
+    public void testSecurityOnTwoBeansInAbsenceOfExplicitSecurityDomain() throws Exception {
+
+        final EjbUnsecuredBean unsecuredBean = InitialContext
+                .doLookup("java:module/" + EjbUnsecuredBean.class.getSimpleName());
+
+        final String echoResult = unsecuredBean.echo("unsecuredBeanEcho");
+        assertEquals("unsecuredBeanEcho", echoResult);
+
+        final Callable<Void> callable = () -> {
+            try {
+                final Principal principal = securedBean.getCallerPrincipal();
+                assertNotNull("EJB must never return a null from the getCallerPrincipal method.", principal);
+                assertEquals("user1", principal.getName());
+            } catch (RuntimeException e) {
+                fail("EJB must provide the caller’s security context information during the execution of a business method (" + e.getMessage() + ")");
+            }
+            return null;
+        };
+        Util.switchIdentitySCF("user1", "password1", callable);
+    }
+
+    public static class DefaultElytronEjbSecurityDomainTestCaseSecuritySetup extends AbstractSecurityDomainSetup {
```

Comment:
This class is being used in the complete wrong way. AbstractSecurityDomainSetup is for security subsystem domains, not elyton ones.

Reply:
+1 @zhantemirov , check https://github.com/wildfly/wildfly/blob/master/testsuite/shared/src/main/java/org/wildfly/test/security/common/elytron/EjbElytronDomainSetup.java ;)